addons/dockable_container/dockable_container.gd

```diff
@@ -195,14 +253,24 @@ func set_layout(value: DockableLayout) -> void:
 		_layout.changed.disconnect(queue_sort)
 	_layout = value
 	_layout.changed.connect(queue_sort)
+	for window in _windows_container.get_children():
+		if not window.name in _layout.windows and window is FloatingWindow:
+			window.prevent_data_erasure = true  # We don't want to delete data.
+			window.close_requested.emit()  # Removes the window.
+			continue
```

Comment:
Isn't this `continue` unnecessary, since it's the last statement of the `for` block?